### DISCUSSION TITLE
render中获取getDvaApp为null，试了好几种方式都不行，请问大家有解决办法吗

### DISCUSSION THREAD

**Discussion by ghost** [umijs/umi#10681] [General]
`export function render(oldRender: any) {
  const app1 = getDvaApp();
  console.log(app1); // null
  getServerMenus().then((res) => {
    if (res.code === RequestCode.Success) {
      const app = getDvaApp();
      console.log('app', app);

        app?._store.dispatch({
          type: 'global/saveState',
          payload: {
            menuData: res.data,
          },
        });  });
  oldRender();
}`

**Comment by ahuigo** [comment]
你可能需要在 config/config.ts 或者　.umirc.ts 中开启一下
```
    export default defineConfig({
        dva:{},
        plugins: [
          '@umijs/plugins/dist/antd',
]
  })
```

**Comment by fz6m** [comment]
See https://github.com/umijs/umi/issues/10680

**Comment by SilentFlute** [comment]
我也在用`dva`, 我这边使用是正常的:

1. `package.json`:
```
//...
"dva": "^2.6.0-beta.22",
//...
```
我记得某个新的版本中`dispatch`返回`Promise`, 但具体哪个版本我忘了, 就直接装这个了

2. 配置中开启一下:
```
//...
dva: {},
//...
```
`umi`中默认有`dva`, 但如果装了, 那会优先使用装的`dva`

3. `import { getDvaApp } from 'umi';`, 然后就能使用`getDvaApp`了
